### PR TITLE
[AMD] Add initial version for run_tests_multi_gpu

### PR DESCRIPTION
**Pull request by mfuntowicz**

### CODE REVIEW

File: .github/workflows/self-push-amd.yml [hunk 1]
```diff
@@ -244,6 +251,106 @@ jobs:
...
+  run_tests_multi_gpu:
+    name: Model tests
+    needs: setup_gpu
+    # `dummy` means there is no test to run
+    if: contains(fromJson(needs.setup.outputs.matrix), 'dummy') != true
+    strategy:
+      fail-fast: false
+      matrix:
+        folders: ${{ fromJson(needs.setup.outputs.matrix) }}
+        machine_type: [ multi-gpu ]
```
Comment (ydshieh) [inline .github/workflows/self-push-amd.yml:263]: I think we can have a single job `run_tests_gpu` but with `machine_type: [single-gpu,multi-gpu ]`.

For current CI, we separate this as they have different docker options: `--gpus 0` and `--gpus all`.

But for this AMD workflow, they are the same.

File: src/transformers/models/bert/modeling_bert.py [hunk 2]
```diff
@@ -251,7 +251,7 @@ def __init__(self, config, position_embedding_type=None):
             )
 
         self.num_attention_heads = config.num_attention_heads
-        self.attention_head_size = int(config.hidden_size / config.num_attention_heads)
+        self.attention_head_size = int(config.hidden_size // config.num_attention_heads)
```
Comment (ydshieh) [inline src/transformers/models/bert/modeling_bert.py:254]: I think you change this just to trigger CI. Should revert before merge.

**Review by ydshieh — COMMENTED** [review]
Thank you! We can simplify the workflow file a  bit.

Also, would love to know how the runners are set up, but it could be an internal documentation.

File: src/transformers/models/bert/modeling_bert.py (hunk 2 above)
Comment (mfuntowicz) [reply src/transformers/models/bert/modeling_bert.py:254]: Yap, it will be reverted for sure, thanks for flagging 👍🏻

Comment (mfuntowicz) [reply src/transformers/models/bert/modeling_bert.py]: Done ✅

File: .github/workflows/self-push-amd.yml (hunk 1 above)
Comment (mfuntowicz) [reply .github/workflows/self-push-amd.yml]: Done ✅

**Review by LysandreJik — APPROVED** [review]
Thanks both for your work on this :)

**Review by ydshieh — APPROVED** [review]
Thanks!

**Pull request merged**